### PR TITLE
feat: Activate/Deactivate ES Consumer

### PR DESCRIPTION
**Pull request by lordmallam**
Implements gather UI and Aether Kernel components of the ticket [AET-521](https://jira.ehealthafrica.org/browse/AET-521)

@sabeen I placed some inline styles based on the [mockups](https://jira.ehealthafrica.org/secure/attachment/20001/SURVEY%20DETAIL%20%20dashboard%20active.png)

### CODE REVIEW

File: app/gather/assets/apps/survey/Survey.jsx [hunk 1]
```diff
@@ -19,24 +19,43 @@
...
 
 const TABLE_VIEW = 'table'
 const SINGLE_VIEW = 'single'
+const DASHBOARD_VIEW = 'dashboard'
 const TABLE_SIZES = [ 10, 25, 50, 100 ]
 
-export default class Survey extends Component {
+const MESSAGES = defineMessages({
+  activate: {
+    id: 'survey.button.actiavte',
```
Comment (admbtlr) [inline app/gather/assets/apps/survey/Survey.jsx]: typo?

File: app/gather/assets/apps/survey/Survey.jsx [hunk 2]
```diff
@@ -19,24 +19,43 @@
...
+    defaultMessage: 'Activate Survey'
+  },
+  deactivate: {
+    id: 'survey.button.deactiavte',
```
Comment (admbtlr) [inline app/gather/assets/apps/survey/Survey.jsx]: same typo again???

File: app/gather/assets/apps/survey/Survey.jsx [hunk 3]
```diff
@@ -46,26 +65,62 @@ export default class Survey extends Component {
...
       allPaths: paths,
-      selectedPaths: paths
+      selectedPaths: paths,
+      isConsumerActive: false,
+      activationError: null
     }
+    this.consumerToggle = this.consumerToggle.bind(this)
+  }
+
+  consumerToggle () {
```
Comment (admbtlr) [inline app/gather/assets/apps/survey/Survey.jsx]: I think `verbNoun` is usually better than `nounVerb` for function names - `toggleConsumer` makes more sense

**Review by admbtlr — CHANGES_REQUESTED** [review]
When I try to submit from ODK Collect with this PR and the develop branch of Aether, I get the error `Instance ID is missing in submission`, which is odd - do you have any idea why? Is that maybe a bug in your last Aether PR?

**Comment by admbtlr** [comment]
False alarm! Looks like this was actually a problem with the form 😅

File: app/gather/api/views.py [hunk 4]
```diff
@@ -17,11 +17,14 @@
...
 
 from rest_framework.viewsets import ModelViewSet
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
 
 from aether.sdk.multitenancy.views import MtViewSetMixin
 
 from .models import Survey, Mask
 from .serializers import SurveySerializer, MaskSerializer
+from ..settings import ES_CONSUMER_URL
```
Comment (obdulia-losantos) [inline app/gather/api/views.py]: The correct way is

```python
from django.conf import settings

# ...

    return Response(settings.ES_CONSUMER_URL)
```

File: app/gather/api/views.py [hunk 5]
```diff
@@ -45,3 +48,8 @@ class MaskViewSet(MtViewSetMixin, ModelViewSet):
     search_fields = ('survey__name', 'name', 'columns',)
     ordering = ('survey', 'name',)
     mt_field = 'survey'
+
+
+@api_view(['GET'])
+def get_es_consumer_url(request, *args, **kwargs):
```
Comment (obdulia-losantos) [inline app/gather/api/views.py]: for this kind of things there is a view in `gather/views.py`  `assets_settings` to return the url.

File: app/gather/assets/apps/utils/settings.spec.jsx [hunk 6]
```diff
@@ -41,14 +41,16 @@ describe('Settings utils', () => {
         .get('/assets-settings')
         .reply(200, {
           aether_apps: ['kernel', 'odk', 'couchdb-sync'],
-          export_max_rows_size: 10000
+          export_max_rows_size: 10000,
+          'es_consumer_url': 'http://localhost:5601/'
```
Comment (obdulia-losantos) [inline app/gather/assets/apps/utils/settings.spec.jsx]: simple is better (you don't need the quotes for the key name) `es_consumer_url: 'http://localhost:5601/'` :wink:

File: app/gather/views.py [hunk 7]
```diff
@@ -28,4 +28,5 @@ def assets_settings(*args, **kwargs):
     return JsonResponse({
         'aether_apps': settings.AETHER_APPS,
         'export_max_rows_size': int(settings.EXPORT_MAX_ROWS_SIZE),
+        'es_consumer_url': settings.ES_CONSUMER_URL,
```
Comment (obdulia-losantos) [inline app/gather/views.py:31]: _Growing Pains_ :see_no_evil:

File: app/gather/api/tests/test_views.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+from django.test import TestCase
```
Comment (obdulia-losantos) [inline app/gather/api/tests/test_views.py]: is this file needed? :thinking:

File: app/gather/assets/apps/survey/SurveyDashboard.jsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
```
Comment (obdulia-losantos) [inline app/gather/assets/apps/survey/SurveyDashboard.jsx:21]: Missing license maybe? :nerd_face:

File: scripts/generate-credentials.sh [hunk 10]
```diff
@@ -135,4 +135,10 @@ DEFAULT_REALM=gather
 REALM_COOKIE=gather-realm
 MULTITENANCY=yes
 # ------------------------------------------------------------------
+
+# ------------------------------------------------------------------
+# Consumer
+# ==================================================================
+ES_CONSUMER_URL=http://gather.local:5601
```
Comment (obdulia-losantos) [inline scripts/generate-credentials.sh:146]: _Growing Pains II part_ :thinking: 

Do we need to set up ES in Gather ??? :scream:

File: app/gather/assets/apps/survey/SurveyDashboard.jsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        <i className='fas fa-pencil-alt invert mr-3' />
+        <FormattedMessage
+          id='survey.no.dashboard.activate'
+          defaultMessage='Activate Dashboard Now' />
+      </button>
+    </div>
+    : <iframe style={{ width: '100%', height: '500px' }} src={props.url} />
+)
+
+export default SurveyDashboard
```
Comment (obdulia-losantos) [inline app/gather/assets/apps/survey/SurveyDashboard.jsx:49]: I don't say this is wrong, but could we stick to only one way of declaring the React Components?

```javascript
import React, { Component } from 'react'

export default class SurveyDashboard extends Component {
  render () {
    // ...
  }
}
```

File: app/gather/assets/apps/survey/Survey.jsx [hunk 12]
```diff
@@ -46,26 +65,62 @@ export default class Survey extends Component {
...
     }
+    this.toggleConsumer = this.toggleConsumer.bind(this)
+  }
+
+  toggleConsumer () {
+    const xform = this.props.xform && this.props.xform.results.length && this.props.xform.results[0]
+    const url = getMappingTopicsAPIPath({ id: xform ? xform.kernel_id : '' })
+
+    // get list of survey topics from kernel
+    fetchUrls([{ name: 'topics', url }])
```
Comment (obdulia-losantos) [inline app/gather/assets/apps/survey/Survey.jsx:80]: To fetch data there is the `FetchUrlsContainer` component to handle all the possible errors and responses, could you use it? 

```javascript
<FetchUrlsContainer
  urls={dashboardUrls}
  targetComponent={SurveyDashboard}
  handleResponse={handleDashboardResponse}
/>
```

Also, could you move all of this to a new component?... You know, I hate the 3000 line files :innocent:

File: scripts/generate-credentials.sh (hunk 10 above)
Comment (lordmallam) [reply scripts/generate-credentials.sh:146]: This is needed to `activate`/`deactivate` the ES consumer

File: app/gather/assets/apps/survey/SurveyDashboard.jsx (hunk 11 above)
Comment (lordmallam) [reply app/gather/assets/apps/survey/SurveyDashboard.jsx:49]: This is a functional component for now

File: scripts/generate-credentials.sh (hunk 10 above)
Comment (obdulia-losantos) [reply scripts/generate-credentials.sh:146]: I knew it, I meant if we need to include all the ES platform to test this new feature... including Kafka&co, aether-producer, es-consumer... :sweat:

File: app/gather/assets/apps/survey/SurveyDashboard.jsx (hunk 11 above)
Comment (obdulia-losantos) [reply app/gather/assets/apps/survey/SurveyDashboard.jsx:49]: :flushed: 

What do you mean?

Comment (lordmallam) [reply app/gather/assets/apps/survey/SurveyDashboard.jsx:49]: I don't need a state right now just render html. Thus, i don't need to extend `component`
